UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,202 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_TRANSFORM_HPP
+#define UNICORN_VIDEO_TRANSFORM_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+
+#define GLM_FORCE_RADIANS
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+/** @brief Base class for every object that can be translated or rotated */
+class Transform
+{
+public:
+    /** @brief Default constructor */
+    Transform();
+
+    /** @brief Default destructor */
+    virtual ~Transform() = default;
+
+    /** @brief Returns true if transformations must be recalculated @sa UpdateModelMatrix, false otherwise */
+    UNICORN_EXPORT bool IsDirty() const;
+
+    /**
+     * @brief Updates orientation with provided direction
+     *
+     * @param[in] direction new direction vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction);
+
+    /**
+     * @brief Generates new orientation from provided direction and up vector
+     *
+     * @param[in] direction new direction vector
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction, glm::vec3 upVector);
+
+    /**
+     * @brief Updates orientation with provided up vector
+     *
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void SetUp(glm::vec3 upVector);
+
+    /**
+     * @brief Sets the world axes and updates orientation
+     *
+     * @param[in]  x world axis
+     * @param[in]  y world axis
+     * @param[in]  z world axis
+     */
+    UNICORN_EXPORT void SetWorldAxes(glm::vec3 x, glm::vec3 y, glm::vec3 z);
+
+    /**
+     * @brief Sets world translation
+     *
+     * @param[in]  translate new translate vector
+     */
+    UNICORN_EXPORT void SetTranslation(glm::vec3 translate);
+
+    /**
+     * @brief Returns direction vector
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
```

Comment:
I suggest moving this `@attention` to setters and other methods that invalidate the state of the object, not getters - it makes more sense to mention this in setters since that's when recalculation/invalidation has to happen.